src/containers/Validators/index.tsx

```diff
@@ -0,0 +1,229 @@
+import React, { useContext, useEffect, useState, useRef } from 'react'
+import axios from 'axios'
+import { useTranslation } from 'react-i18next'
+import { useQuery } from 'react-query'
+import NoMatch from '../NoMatch'
+import Loader from '../shared/components/Loader'
+import { Tabs } from '../shared/components/Tabs'
+import {
+  analytics,
+  ANALYTIC_TYPES,
+  FETCH_INTERVAL_VHS_MILLIS,
+  NOT_FOUND,
+  SERVER_ERROR,
+} from '../shared/utils'
+import { getLedger } from '../../rippled'
+import SimpleTab from './SimpleTab'
+import { HistoryTab } from './HistoryTab'
+import './validator.scss'
+import SocketContext from '../shared/SocketContext'
+
+const ERROR_MESSAGES = {
+  [NOT_FOUND]: {
+    title: 'validator_not_found',
+    hints: ['check_validator_key'],
+  },
+  default: {
+    title: 'generic_error',
+    hints: ['not_your_fault'],
+  },
+}
+
+const getErrorMessage = (error: keyof typeof ERROR_MESSAGES | null) =>
+  (error && ERROR_MESSAGES[error]) || ERROR_MESSAGES.default
+
+interface Props {
+  match: {
```

Comment:
You can use `useRouteMatch` and `useParams` to get these properties.

https://v5.reactrouter.com/web/api/Hooks/useroutematch
https://v5.reactrouter.com/web/api/Hooks/useparams